plugins-compat-tester/src/main/java/org/jenkins/tools/test/PluginCompatTester.java

```diff
@@ -658,10 +659,9 @@ private void addSplitPluginDependencies(String thisPlugin, MavenRunner.Config mc
                     System.out.println("Skipping implicit dep " + thisPlugin + " → " + plugin);
                     continue;
                 }
-                VersionNumber splitPoint = new VersionNumber(pieces[1]);
+                String splitPoint = pieces[1];
                 VersionNumber declaredMinimum = new VersionNumber(pieces[2]);
-                // TODO this should only happen if the tested core version is ≥ splitPoint
-                if (coreDep.compareTo(splitPoint) <= 0 && !pluginDeps.containsKey(plugin)) {
+                if (coreCoordinates.compareVersionTo(splitPoint) >= 0 && !pluginDeps.containsKey(plugin)) {
```

Comment:
I think you should still maintain the previous condition. Even though you check the tested version is bigger than the splitPoint, the plugin should be prepared for that if its baseline is bigger too.